packages/forms/src/directives/validators.ts

```diff
@@ -656,8 +664,12 @@ export class MaxLengthValidator implements Validator, OnChanges {
   }
 
   private _createValidator(): void {
-    this._validator = maxLengthValidator(
-        typeof this.maxlength === 'number' ? this.maxlength : parseInt(this.maxlength, 10));
+    this._validator =
+        this.enabled() ? maxLengthValidator(toInteger(this.maxlength!)) : nullValidator;
+  }
+
+  enabled(): boolean {
+    return this.maxlength !== null;
```

Comment:
Similar to the minlength validator comment above:
```suggestion
    return this.maxlength != null /* both `null` and `undefined` */;
```